pkg/portal/ssh/ssh.go

```diff
@@ -159,29 +168,58 @@ func (s *SSH) New(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
-	port := ""
-	if s.env.IsLocalDevelopmentMode() {
-		port = "-p 2222 "
-	}
-
-	s.sendResponse(w, &response{
-		Command:  fmt.Sprintf("ssh %s%s@%s", port, username, host),
-		Password: password,
-	})
+	s.sendResponse(w, host, username, password, "", s.env.IsLocalDevelopmentMode())
 }
 
-func (s *SSH) sendResponse(w http.ResponseWriter, resp *response) {
-	b, err := json.MarshalIndent(resp, "", "    ")
+func (s *SSH) sendResponse(w http.ResponseWriter, hostname, username, password, error string, isLocalDevelopmentMode bool) {
+	w.Header().Set("Content-Type", "application/json")
+
+	enc := json.NewEncoder(w)
+	enc.SetEscapeHTML(false)
+	enc.SetIndent("", "    ")
+
+	if error != "" {
+		err := enc.Encode(response{Error: error})
+		if err != nil {
+			s.internalServerError(w, err)
+		}
+		return
+	}
+	fingerprint := cryptossh.FingerprintSHA256(s.hostPubKey)
+	command, err := createLoginCommand(isLocalDevelopmentMode, username, hostname, fingerprint)
+	resp := response{Command: command, Password: password}
+	if err != nil {
+		s.internalServerError(w, err)
+	}
+	err = enc.Encode(resp)
 	if err != nil {
 		s.internalServerError(w, err)
-		return
 	}
-
-	w.Header().Set("Content-Type", "application/json")
-	_, _ = w.Write(b)
 }
 
 func (s *SSH) internalServerError(w http.ResponseWriter, err error) {
 	s.log.Warn(err)
 	http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
 }
+
+const (
+	sshCommand = "echo '{{ .Hostname }} ssh-rsa {{.PublicKeyFingerprint}}' > {{.Hostname}}_known_host ; " +
+		"ssh -o UserKnownHostsFile={{.Hostname}}_known_host{{if .IsLocalDevelopmentMode}} -p 2222{{end}} {{.User}}@{{.Hostname}}"
+)
+
+func createLoginCommand(IsLocalDevelopmentMode bool, user, host, fingerPrint string) (string, error) {
+	tmp := template.New("command")
+	tmp, err := tmp.Parse(sshCommand)
+	if err != nil {
+		return "", err
+	}
+	type fields struct {
+		User                   string
+		Hostname               string
+		PublicKeyFingerprint   string
+		IsLocalDevelopmentMode bool
+	}
+	var buff bytes.Buffer
+	err = tmp.Execute(&buff, fields{user, host, fingerPrint, IsLocalDevelopmentMode})
```

Comment:
why do we call Execute during command creation?

Reply:
when do you want to call it? the username is unique to each request, so we cannot just run it once.

Reply:
Oh I think I understand the confusion now, I'm not executing the command, I'm executing the text template

Reply:
aye, thanks for clearing that up.